state/indexer/sink/psql/psql_test.go

```diff
@@ -0,0 +1,365 @@
+	sqlStmt := sq.
+		Select("tx_result", fmt.Sprintf("%s.id", TableResultTx), "tx_result_id", "hash", "chain_id").
+		Distinct().From(TableResultTx).
+		InnerJoin(join).
+		Where(fmt.Sprintf("hash = $1 AND chain_id = '%s'", chainID), fmt.Sprintf("%X", hash))
+
+	rows, err := sqlStmt.RunWith(db).Query()
+	if err != nil {
+		return nil, err
+	}
+
+	defer rows.Close()
+
+	if rows.Next() {
+		var txResult []byte
+		var txResultID, txid int
+		var h, cid string
+		err = rows.Scan(&txResult, &txResultID, &txid, &h, &cid)
+		if err != nil {
+			return nil, nil
+		}
+
+		msg := new(abci.TxResult)
+		err = proto.Unmarshal(txResult, msg)
+		if err != nil {
+			return nil, err
+		}
+
+		return msg, err
+	}
+
+	// No result
+	return nil, nil
+}
+
+func verifyTimeStamp(tb string) error {
+
+	// We assume the tx indexing time would not exceed 2 second from now
+	sqlStmt := sq.
+		Select(fmt.Sprintf("%s.created_at", tb)).
+		Distinct().From(tb).
+		Where(fmt.Sprintf("%s.created_at >= $1", tb), time.Now().Add(-2*time.Second))
+
+	rows, err := sqlStmt.RunWith(db).Query()
+	if err != nil {
+		return err
+	}
+
+	defer rows.Close()
+
+	if rows.Next() {
+		var ts string
+		err = rows.Scan(&ts)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	return errors.New("no result")
+}
+
+func verifyBlock(h int64) (bool, error) {
+	sqlStmt := sq.
+		Select("height").
+		Distinct().
+		From(TableEventBlock).
+		Where(fmt.Sprintf("height = %d", h))
+	rows, err := sqlStmt.RunWith(db).Query()
+	if err != nil {
+		return false, err
+	}
+
+	defer rows.Close()
+
+	if !rows.Next() {
+		return false, nil
+	}
+
+	sqlStmt = sq.
+		Select("type, height", "chain_id").
+		Distinct().
+		From(TableEventBlock).
+		Where(fmt.Sprintf("height = %d AND type = '%s' AND chain_id = '%s'", h, types.EventTypeBeginBlock, chainID))
+
+	rows, err = sqlStmt.RunWith(db).Query()
+	if err != nil {
+		return false, err
+	}
+
+	if !rows.Next() {
+		return false, nil
+	}
+
+	sqlStmt = sq.
+		Select("type, height").
+		Distinct().
+		From(TableEventBlock).
+		Where(fmt.Sprintf("height = %d AND type = '%s'", h, types.EventTypeEndBlock))
+	rows, err = sqlStmt.RunWith(db).Query()
+
+	if err != nil {
+		return false, err
+	}
+
+	return rows.Next(), nil
+}
+
+func setupDB(t *testing.T) (*dockertest.Pool, error) {
+	t.Helper()
+	pool, err := dockertest.NewPool(os.Getenv("DOCKER_URL"))
+
+	assert.NoError(t, err)
+
+	resource, err = pool.RunWithOptions(&dockertest.RunOptions{
+		Repository: DriverName,
+		Tag:        "13",
+		Env: []string{
+			"POSTGRES_USER=" + user,
+			"POSTGRES_PASSWORD=" + password,
+			"POSTGRES_DB=" + dbName,
+			"listen_addresses = '*'",
+		},
+		ExposedPorts: []string{port},
+	}, func(config *docker.HostConfig) {
+		// set AutoRemove to true so that stopped container goes away by itself
+		config.AutoRemove = true
+		config.RestartPolicy = docker.RestartPolicy{
+			Name: "no",
+		}
+	})
+
+	assert.NoError(t, err)
```

Comment:
we should almost certianly be using require rather than assert for this/most things.